Conflicting implementations of trait `platform::platform::x11::util::format::Formattable` for type `u8`

```
error[E0119]: conflicting implementations of trait `platform::platform::x11::util::format::Formattable` for type `u8`:
  --> /home/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.16.0/src/platform/linux/x11/util/format.rs:54:1
   |
53 | impl Formattable for c_char { const FORMAT: Format = Format::Char; }
   | --------------------------- first implementation here
54 | impl Formattable for c_uchar { const FORMAT: Format = Format::Char; }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `u8`
```
In https://github.com/servo/servo/pull/21087#issuecomment-401608678 on targets `arm-unknown-linux-gnueabihf` and `aarch64-unknown-linux-gnu`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 
 - Added logging through `log`. Logging will become more extensive over time.
 - On X11 and Windows, the window's DPI factor is guessed before creating the window. This *greatly* cuts back on unsightly auto-resizing that would occur immediately after window creation.
+- Fixed X11 backend compilation for environments where `c_char` is unsigned.
 
 # Version 0.16.0 (2018-06-25)
 

--- a/src/platform/linux/x11/util/format.rs
+++ b/src/platform/linux/x11/util/format.rs
@@ -50,7 +50,7 @@ pub trait Formattable: Debug + Clone + Copy + PartialEq + PartialOrd {
 }
 
 // You might be surprised by the absence of c_int, but not as surprised as X11 would be by the presence of it.
-impl Formattable for c_char { const FORMAT: Format = Format::Char; }
+impl Formattable for c_schar { const FORMAT: Format = Format::Char; }
 impl Formattable for c_uchar { const FORMAT: Format = Format::Char; }
 impl Formattable for c_short { const FORMAT: Format = Format::Short; }
 impl Formattable for c_ushort { const FORMAT: Format = Format::Short; }

